core/tests/snapshots.rs

```diff
@@ -255,14 +257,14 @@ mod tests {
             snapshot_request_receiver,
             accounts_package_sender,
         };
-        for slot in 0..last_slot {
-            let mut bank = Bank::new_from_parent(&bank_forks[slot], &Pubkey::default(), slot + 1);
+        for slot in 1..=last_slot {
+            let mut bank = Bank::new_from_parent(&bank_forks[slot - 1], &Pubkey::default(), slot);
             f(&mut bank, mint_keypair);
             let bank = bank_forks.insert(bank);
             // Set root to make sure we don't end up with too many account storage entries
             // and to allow snapshotting of bank and the purging logic on status_cache to
             // kick in
-            if slot % set_root_interval == 0 || slot == last_slot - 1 {
+            if slot % set_root_interval == 0 || slot == last_slot {
```

Comment:
Had to change around the slot numbers so they were proper multiples of the snapshot intervals. Previously the check on line 267 (265 old) was true on `slot = interval + 1`, which mean the bank snapshot would never be sent to be archived.